### PR TITLE
feat: logout, pem, seed, account menu

### PR DESCRIPTION
**Pull request by nblenke**
Various updates around authentication

## Description

- Add account menu
- Add log in/log out
- Add create or use seed phrase
- Add import pem
- Refactor header to use grid
- Fix heading alignment on home

## Screenshots (if applicable)
Screen Shot 2023-09-10 at 2 17 15 PM

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [x] I have added or updated tests and they pass.
- [x] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

**Review by fmorency — CHANGES_REQUESTED** [review]
Very nice, thanks!

Some comments
- Display the beginning and the end of the MANY address like displayed in https://github.com/liftedinit/ghostcloud/issues/14#issuecomment-1707370640
- Always display the "Account" icon in the top menu and offer to log in from it if we're logged out. You could even display the `Dashboard` entry from that menu only when logged in and removed it from the header—food for thought.
- The social login option doesn't result in the same MANY address as in Alberto and Gwen. We need to fix this.
  - Log in on https://alberto.app using your social account. Notice your MANY address
  - Log in on https://gwen.liftedinit.app using your social account. Notice that your MANY address is the same as the one in Alberto
  - Log in on Ghostcloud using your social account. Notice that your MANY address is different than from Alberto and Gwen.

Let me know if I need to modify anything from our Web3Auth account.

File: src/features/accounts/components/add-account-modal/add-account-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+export function AddAccountModal({
+  isOpen,
+  onClose,
+}: {
+  isOpen: boolean
+  onClose: () => void
+}) {
+  const [addMethod, setAddMethod] = React.useState<AddMethodState>('')
+  const [showDefaultFooter, setShowDefaultFooter] =
```
Comment (fmorency) [inline src/features/accounts/components/add-account-modal/add-account-modal.tsx]:
```
src/features/accounts/components/add-account-modal/add-account-modal.tsx
  Line 42:29:  'setShowDefaultFooter' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

**Comment by nblenke** [comment]
> * Always display the "Account" icon in the top menu and offer to log in from it if we're logged out. You could even display the `Dashboard` entry from that menu only when logged in and removed it from the header—food for thought.

We can do that. Is the expectation that the "Get Started" CTA on the homepage would launch the login modal? And the user would be redirected to /dashboard on success?

> * The social login option doesn't result in the same MANY address as in Alberto and Gwen. We need to fix this.
>   
>   * Log in on https://alberto.app using your social account. Notice your MANY address
>   * Log in on https://gwen.liftedinit.app using your social account. Notice that your MANY address is the same as the one in Alberto
>   * Log in on Ghostcloud using your social account. Notice that your MANY address is different than from Alberto and Gwen.
> 
> Let me know if I need to modify anything from our Web3Auth account.

I don't see anything that looks unique here other than the `GHOSTCLOUD.ACCOUNTS` and `GHOSTCLOUD.IDENTITIES` values in the store. This may be something that needs to be configured in the WebAuth3 admin? I'm not sure.

**Comment by fmorency** [comment]
> > * Always display the "Account" icon in the top menu and offer to log in from it if we're logged out. You could even display the `Dashboard` entry from that menu only when logged in and removed it from the header—food for thought.
> 
> We can do that. Is the expectation that the "Get Started" CTA on the homepage would launch the login modal? And the user would be redirected to /dashboard on success?

Something like that, yes. As long as the flow makes sense and minimizes the number of clicks. I would like icons and flow to be consistent.
> 
> > * The social login option doesn't result in the same MANY address as in Alberto and Gwen. We need to fix this.
> >   
> >   * Log in on https://alberto.app using your social account. Notice your MANY address
> >   * Log in on https://gwen.liftedinit.app using your social account. Notice that your MANY address is the same as the one in Alberto
> >   * Log in on Ghostcloud using your social account. Notice that your MANY address is different than from Alberto and Gwen.
> > 
> > Let me know if I need to modify anything from our Web3Auth account.
> 
> I don't see anything that looks unique here other than the `GHOSTCLOUD.ACCOUNTS` and `GHOSTCLOUD.IDENTITIES` values in the store. This may be something that needs to be configured in the WebAuth3 admin? I'm not sure.

I can take a look. I recall we had this issue when trying to upgrade our packages to node 18.

**Review by fmorency — APPROVED** [review]

**Comment by fmorency** [comment]
Thanks! That's perfect for this PR. We can refine it later.

**Comment by fmorency** [comment]
> > > * Always display the "Account" icon in the top menu and offer to log in from it if we're logged out. You could even display the `Dashboard` entry from that menu only when logged in and removed it from the header—food for thought.
> > 
> > 
> > We can do that. Is the expectation that the "Get Started" CTA on the homepage would launch the login modal? And the user would be redirected to /dashboard on success?
> 
> Something like that, yes. As long as the flow makes sense and minimizes the number of clicks. I would like icons and flow to be consistent.
> 
> > > * The social login option doesn't result in the same MANY address as in Alberto and Gwen. We need to fix this.
> > >   
> > >   * Log in on https://alberto.app using your social account. Notice your MANY address
> > >   * Log in on https://gwen.liftedinit.app using your social account. Notice that your MANY address is the same as the one in Alberto
> > >   * Log in on Ghostcloud using your social account. Notice that your MANY address is different than from Alberto and Gwen.
> > > 
> > > Let me know if I need to modify anything from our Web3Auth account.
> > 
> > 
> > I don't see anything that looks unique here other than the `GHOSTCLOUD.ACCOUNTS` and `GHOSTCLOUD.IDENTITIES` values in the store. This may be something that needs to be configured in the WebAuth3 admin? I'm not sure.
> 
> I can take a look. I recall we had this issue when trying to upgrade our packages to node 18.

For the sake of posterity, this issue comes from the differences between the staging vs production build. See https://github.com/liftedinit/ghostcloud/issues/20

**Pull request merged**